src/utils/request.ts

```diff
@@ -0,0 +1,14 @@
+import Authentication from "../config/auth";
+
+type FetchParams = Parameters<typeof fetch>;
+
+export async function authFetch(url: FetchParams[0], options: FetchParams[1]): Promise<Response>{
+    const tokenResponse = await Authentication.getToken();
+    return await fetch(url, {
+        ...options,
+        headers: {
+            'Content-Type': 'application/json',
```

Comment:
This forces all requests to be in JSON, not that we're planning on using another serialization format but just wanted to point that out 😆.

On another note, you generally don't want to override _all_ the headers in the `options` parameter. The caller should be able to add additional headers, with maybe only the `Authorization` header being overriden by this function.

Reply:
Good to know and I agree. It is best practice to let other developers override 'Content-Type' and other headers.